### PR TITLE
Add test case for a pod becoming schedulable when another pod is deleted

### PR DESCRIPTION
**Pull request by nodo**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
It adds an integration test for a pod deletion event. This is part of an effort to add more coverage to event handlers. In this case, the pod deletion event triggers the scheduler to assign the new pod to a node.

**Which issue(s) this PR fixes**:
Part of #91111. However, the issue is not fixed yet, I am still planning to add a few more tests.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: test/integration/scheduler/predicates_test.go [hunk 1]
```diff
@@ -1074,14 +1075,40 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
...
 				if err := testutils.RemoveTaintOffNode(cs, "node-tainted", taint); err != nil {
 					return fmt.Errorf("cannot remove taint off node: %v", err)
 				}
 				return nil
 			},
 		},
+		{
+			name: "pod gets deleted",
+			init: func(cs kubernetes.Interface, ns string) error {
+				resourceList := &v1.ResourceList{
```
Comment (alculquicondor) [inline test/integration/scheduler/predicates_test.go:1089]: s/resourceList/nodeResources

**Review by alculquicondor — COMMENTED** [review]
/approve

from scheduler

File: test/integration/scheduler/predicates_test.go (hunk 1 above)
Comment (nodo) [reply test/integration/scheduler/predicates_test.go]: Thanks, I have renamed it 👍

File: test/integration/scheduler/predicates_test.go [hunk 2]
```diff
@@ -1074,14 +1075,40 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
...
+			update: func(cs kubernetes.Interface, _ string) error {
 				taint := v1.Taint{Key: "test", Value: "test", Effect: v1.TaintEffectNoSchedule}
 				if err := testutils.RemoveTaintOffNode(cs, "node-tainted", taint); err != nil {
 					return fmt.Errorf("cannot remove taint off node: %v", err)
 				}
 				return nil
 			},
 		},
+		{
+			name: "pod gets deleted",
```
Comment (alculquicondor) [inline test/integration/scheduler/predicates_test.go:1087]: other pod gets deleted.
Sorry I missed this before

Comment (nodo) [reply test/integration/scheduler/predicates_test.go:1087]: no worries! makes sense, I will add it

**Comment by nodo** [comment]
Thanks @alculquicondor !

**Pull request merged**